Guide.docc/RuntimeBehavior.md

```diff
@@ -0,0 +1,207 @@
+# Runtime Behavior
+
+
+Learn how Swift concurrency runtime semantics differ from other runtimes you may 
+be familiar with, and familiarize yourself with common patterns to achieve 
+similar end results in terms of execution semantics.
+
+Swift's concurrency model with a strong focus on async/await, actors and tasks,
+means that some patterns from other libraries or concurrency runtimes don't 
+translate directly into this new model. In this section, we'll explore common 
+patterns and differences in runtime behavior to be aware of, and how to address 
+them while you migrate your code to Swift concurrency.
+
+### Ordered stream processing
+
+### Ordered work processing in actors, when enqueueing from a synchronous contexts
+
+Swift concurrency naturally enforces program order for asynchronous code as long
+as the execution remains in a single Task - this is equivalent to using "a single
+thread" to execute some work, but is more resource efficient because the task may
+suspend while waiting for some work, for example:
+
+```
+// ✅ Guaranteed order, since caller is a single task
+let printer: Printer = ...
+await printer.print(1)
+await printer.print(2)
+await printer.print(3)
+```
+
+This code is structurally guaranteed to execute the prints in the expected "1, 2, 3"
+order, because the caller is a single task. Things
+
+Dispatch queues offered the common `queue.async { ... }` way to kick off some
+asynchronous work without waiting for its result. In dispatch, if one were to
+write the following code:
+
+```swift
+let queue = DispatchSerialQueue(label: "queue")
+
+queue.async { print(1) }
+queue.async { print(2) }
+queue.async { print(3) }
+```
+
+The order of the elements printed is guaranteed to be `1`, `2` and finally `3`.
+
+At first, it may seem like `Task { ... }` is exactly the same, because it also
+kicks off some asynchronous computation without waiting for it to complete.
+A naively port of the same code might look like this:
+
+```swift
+// ⚠️ any order of prints is expected
+Task { print(1) }
+Task { print(2) }
+Task { print(3) }
+```
+
+This example **does not** guarantee anything about the order of the printed values,
+because Tasks are enqueued on a global (concurrent) threadpool which uses multiple
```

Comment:
It would help if we spoke about executors here first IMO.

Reply:
There's sections to be written about executors, that'd address this concern imho; it would not be in this section anyway